inginious/frontend/pages/admin/admin.py

```diff
@@ -23,15 +24,34 @@ def POST_AUTH(self, *args, **kwargs):
 
     def show_page(self):
         """Display page"""
-
-        page = int(request.form.get("page")) if request.form.get("page") is not None else 1
-        user_per_page = 10  # TODO probably better to let user define user_per_page
-        all_users = self.user_manager.get_users_info(usernames=None, limit=user_per_page, skip=(page-1)*user_per_page)
-        size_users = self.user_manager.get_users_count()
-        pages = size_users // user_per_page + (size_users % user_per_page > 0) if user_per_page > 0 else 1
+        user_input = request.form
+        if "displayed_selection" in user_input:
+            # Mostly happened when changing page
+            params = json.loads(user_input.get("displayed_selection", ""))
+        else:
+            params={"limit":10,"sort_by":"username","order":1}
+            for key in ["username", "realname", "email"]:
+                params[key] = user_input[key] if key in user_input else ""
+            if "activated" in user_input:
+                params["activated"] = user_input["activated"]
+            if "sort_by" in user_input and "order" in user_input:
+                params["sort_by"]=user_input["sort_by"]
+                params["order"]= int(user_input["order"])
+            if "limit" in user_input:
+                params["limit"] = int(user_input["limit"])
+        query = {}
+        for key in ["username","realname","email"]:
+            query[key]={'$regex': f".*{params[key]}.*"}
+        if 'activated' in params:
+            query["activate"] = {'$exists': not params["activated"]}
+        usernames = [x['username'] for x in
+                     self.database.users.find(query)]
+        page = int(user_input.get("page")) if user_input.get("page") is not None else 1
+        all_users = self.user_manager.get_users_info(usernames=usernames, limit=params["limit"], skip=(page-1)*params["limit"],sort_key=params["sort_by"],order=params["order"])
```

Comment:
Please, please read the code of that function and realize it does the same job as what you just did and then threw away, that is, fetching all user data from database `users` collection according to ome criteria.